### DISCUSSION TITLE
Looking for best practice using J2CL together with Typescript to migrate Java piece by piece to the web

### DISCUSSION THREAD

**Discussion by JuergenMutschall** [treblereel/Beginning_J2CL#3] [Q&A] [answered]
Hi,

I am following the gitter discussion about j2cl for a long time.  Still a long way to go.
But maybe now is the time to start using j2cl to use java and typescript together.

I have a big eclipse java source base, I would like to migrate to js Libs used in node/angular/Typescript.

I tried your gwt processor annotations, but your example does not compile.

What is your recommendation to start with, if i want to migrate multiple libraries to the web, e.g. starting with a small text processing library, which can be debugged in a node/angular browser context with source maps?

Regards,

Juergen

**Comment by treblereel** [comment, marked as answer]
@JuergenMutschall i hope, i fixed the issue.

I think you can start the same conversation at https://github.com/Vertispan/j2clmavenplugin/discussions simply because of bigger audience.

Without details about your projects, it's very difficult to say something. If i were you i would split it into logical, replaceble blocks. I and migrate it one by one. 

'eclipse java source' - i used to port eclipse bpmn definitions, it wasn't simple, because it was converted from C code to Java. So it depends ...
